### DISCUSSION TITLE
How to configure DE shortcut?

### DISCUSSION THREAD

**Discussion by beedell-roke** [quexten/goldwarden#42] [Q&A] [answered]
> ![20240103T171035GMT](https://github.com/quexten/goldwarden/assets/42837531/85bcdc72-583f-4a00-94be-53a2c50bf14d)

Where in https://github.com/quexten/goldwarden#setup-and-usage is it explained? I don't see it at https://github.com/quexten/goldwarden/wiki/Getting-Started.

**Comment by quexten** [comment, marked as answer]
Thanks for the question. First of all, is the screenshot the way the app looks for you (with the double window borders?) if so that's a bug to investigate. In that case, let me know your distro version and desktop environment.

To detect system wide hotkeys in flatpak, a portal is needed. This is not supported on all DE's yet. Instead, you can configure a shortcut in your DE to call a command which talks to goldwarden over dbus.

Described here, with commands for Gnome: https://github.com/quexten/goldwarden/wiki/System%E2%80%90wide-Autotype
For KDE, this guide should help: https://docs.kde.org/stable5/en/khelpcenter/fundamentals/shortcuts.html

Once the global shortcut portal is supported, this will hopefully become more seamless :)

**Reply by beedell-roke** [reply]
> https://github.com/quexten/goldwarden/discussions/42#discussioncomment-8005398

@quexten,

1.	> Thanks for the question. First of all, is the screenshot the way the app looks for you (with the double window borders?) if so that's a bug to investigate. In that case, let me know your distro version and desktop environment.
	
	See https://bugs.kde.org/show_bug.cgi?id=479372, especially its https://bugs.kde.org/page.cgi?id=fields.html#see_also. I'm using KWin with `noborderrule=2` with [cpe:2.3:a:kde:plasma:5.27.10](https://nvd.nist.gov/products/cpe/search/results?namingFormat=2.3&orderBy=2.3&keyword=cpe%3A2.3%3Aa%3Akde%3Aplasma&status=FINAL%2CDEPRECATED) on [cpe:/o:opensuse:tumbleweed:20231226](https://nvd.nist.gov/products/cpe/search/results?namingFormat=2.3&orderBy=2.3&keyword=cpe%3A2.3%3Aa%3Aopensuse%3Atumbleweed&status=FINAL).
	
1.	> To detect system wide hotkeys in flatpak, a portal is needed. This is not supported on all DE's yet. Instead, you can configure a shortcut in your DE to call a command which talks to goldwarden over dbus.
	
	How might I ascertain whether Plasma supports this? Do you happen to know (so that I need not configure DBus)?

1.	<strike>

	<blockQuote>
	
	Described here, with commands for Gnome: [Wiki: System‐wide Autotype ()](https://github.com/quexten/goldwarden/wiki/System%E2%80%90wide-Autotype)\
	For KDE, this guide should help: [docs.kde.org/stable5/en/khelpcenter/fundamentals/shortcuts.html](https://docs.kde.org/stable5/en/khelpcenter/fundamentals/shortcuts.html)
	
	</blockQuote>

	I am thankful. I'll look. However, regardless, is this actually explained in https://github.com/quexten/goldwarden/blob/main/Readme.md somewhere? I ask because the application does state so.

	</strike>

	Sorry. Was being slow there.

**Reply by quexten** [reply]
Thanks for pointing out the kde/tumbleweed bug. I had only tested KDE on Kubuntu previously, there it seemed to work fine on XOrg and (after a small tweak) fine on Waylaynd, but since it seems to be an upstream bug, not too much I can do there.

> How might I ascertain whether Plasma supports this? Do you happen to know (so that I need not configure DBus)

I think the KDE portal actually already supports global hotkeys, but since it's *only* KDE and not wl-roots or Gnome yet, I have not yet implemented this part in Goldwarden. So for now, adding a custom shortcut with the dbus command is required. Though in KDE it should only be a few clicks. I was planning to look at this after at least Gnome also implements it https://gitlab.gnome.org/GNOME/xdg-desktop-portal-gnome/-/issues/47 , https://github.com/quexten/goldwarden/issues/39.

Technically, on XOrg, its also feasible for the app to just act as a keylogger and inspect all keystrokes - even in the "sandbox" of a Flatpak, but I'd really be happier if it's done properly via the upcoming portal.

**Reply by beedell-roke** [reply]
> https://github.com/quexten/goldwarden/discussions/42#discussioncomment-8006994

@quexten,

> Thanks for pointing out the kde/tumbleweed bug. I had only tested KDE on Kubuntu previously, there it seemed to work fine on XOrg and (after a small tweak) fine on Waylaynd, but since it seems to be an upstream bug, not too much I can do there.

You need to enable the `noborderrule=2` preference by setting

> ![image](https://github.com/quexten/goldwarden/assets/42837531/2a6a31c5-126f-4541-a972-3cd8a0163ad9)

which you need to be using X11 for (due to https://bugs.kde.org/show_bug.cgi?id=470535#c0).
